pkg/storage/s3/s3.go

```diff
@@ -106,14 +97,26 @@ func (d *driver) createBucket(svc *s3.S3) error {
 }
 
 func (d *driver) CompleteConfiguration() error {
+	s3cfg, err := util.GetConfig(d.Name, d.Namespace)
+	if err != nil {
+		return err
+	}
 	sess, err := session.NewSession(&aws.Config{
-		Credentials: credentials.NewStaticCredentials("", "", ""),
+		Credentials: credentials.NewStaticCredentials(s3cfg.Storage.S3.AccessKey, s3cfg.Storage.S3.SecretKey, ""),
+		Region:      &s3cfg.Storage.S3.Region,
 	})
 	if err != nil {
 		return err
 	}
 	svc := s3.New(sess)
 
+	if len(d.Config.Bucket) == 0 {
+		d.Config.Bucket = s3cfg.Storage.S3.Bucket
+	}
+	if len(d.Config.Region) == 0 {
+		d.Config.Region = s3cfg.Storage.S3.Region
+	}
+
```

Comment:
comment on the bucket creation logic below:

You need to ensure that on re-sync, we are making sure the bucket exists.  That means a few things:
1) Adding a condition on the registry resource that indicates when we have successfully created the bucket (we should not be deploying the registry until we are sure we created the bucket)
2) a mechanism for ensuring that if the S3 bucket name changed, the condition is cleared until we create the new bucket.

this can be a follow up but i don't consider the s3 feature complete until we have these things.